File: leetcode_ly/java/Binary Search/ValidPerfectSquare.java
public class Solution {
    public boolean isPerfectSquare(int num) {
        int i = 1;
        int j = num;
        while(i <= j){
            int m = i + (j - i) / 2;
            if((long)m * m == (long)num) return true;
            else if((long)m * m < (long)num) i = m + 1;
            else j = m - 1;
        }
        return false;
    }
}

File: leetcode_ly/c++/BinarySearch/ValidPerfectSquare.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        int low = 1;
        int high = num;
        while(low <= high){
            long middle = low + (high - low) / 2;
            if(middle * middle == num) return true;
            else if(middle * middle > num) high = middle - 1;
            else low = middle + 1;
        }
        return false;
    }
};